kaltura_admin.py

```diff
@@ -34,74 +38,100 @@ def get_kaltura_session(user_id, partnerID, adminkey):
-    result = client.user.list(filter, pager)
-    log.info("Retriving kmc: {}".format(i))
-    log.info("Total Records: {}".format(result.totalCount))
-    for item in result.objects:
-        if item.roleNames == "Publisher Administrator" or item.roleNames == "Content Manager":
-            # remove @xxxxx
-            if '@' in item.id:
-                item_id = item.id.partition('@')
-                id = item_id[0]
-                username_list.append(id)
-            else:
-                username_list.append(item.id)
-# remove duplicate
-result = list(set(username_list))
-
-# sort list
-result.sort()
-
-# result
-df = DataFrame(result)
-
-# generate the output csv file with timestamp
-filename = 'Kaltura_admin_{}.csv'.format(date.today())
-df.to_csv(filename, index=False, header=False)
-
-"""**Connect to filezilla**"""
-cnopts = pysftp.CnOpts()
-cnopts.hostkeys = None
-# connect to the sftp server and authenticate with a private key
-sftp = pysftp.Connection(
-    host=ftp_server["host"],
-    username=ftp_server["username"],
-    password=ftp_server["password"],
-    cnopts=cnopts)
-
-# send file to SFTP server
-log.info(
-    f'''Uploaded file to sftp server. File status: {sftp.put(filename)}''')
+
+def sftp_kmc_admin_list(kmc_admin_df, ftp_server):
+    """
+    sftp the KMC admin user information as an csv file to SFTP server
+    """
+    # connect to the sftp server and authenticate with a private key
+    cnopts = pysftp.CnOpts()
+    cnopts.hostkeys = None
+    sftp = pysftp.Connection(
+        host=ftp_server["host"],
+        username=ftp_server["username"],
+        password=ftp_server["password"],
+        cnopts=cnopts)
+
+    # send file to SFTP server
+    filename = 'Kaltura_admin_{}.csv'.format(date.today())
+    with sftp.open(filename, 'w+') as f:
+        chunksize = 100
+        with tqdm(total=len(kmc_admin_df)) as progbar:
+            log.info(f'''status {progbar}''')
+            kmc_admin_df.to_csv(f, index=False, chunksize=chunksize)
+            progbar.update(chunksize)
+    log.info(
+        f'''Uploaded file to sftp server.''')
+
+
+def main():
+    """
+    Get the configuration file
+    """
+    parser = argparse.ArgumentParser(
+        description='Generate Kaltura KMC admin list')
+    parser.add_argument(
+        'env', type=str, help='env json file with kmc settings')
+    args = parser.parse_args()
+    if not os.path.isfile(args.env):
+        log.warn("env.json file does not exist. Please provide file path.")
+        sys.exit()
+
+    """
+    Generate admin list and report
+    """
+    # load the configuration file
+    with open(args.env, "r") as jsonfile:
+        data = json.load(jsonfile)
+        kmcs = data["kmcs"]
+        kaltura_user_id = data["user_id"]
+        ftp_server = data["ftp_server"]
+
+    # generate the output csv file with timestamp
+    kmc_admin_df = get_kmc_admin_list(kaltura_user_id, kmcs)
+    log.info(f'''shape of kmc_admin_list {kmc_admin_df.shape}''')
```

Comment:
nitpick: This doesn't need 3 single quotes

Reply:
replaced 3 single quotes with single "